core/src/main/scala/org/http4s/Static.scala

```diff
@@ -0,0 +1,212 @@
+package org.http4s
+
+
+import java.io._
+import java.net.URL
+
+
+import cats.implicits.{catsSyntaxEither => _, _}
+import cats.data.OptionT
+import cats.effect.{ContextShift, Sync}
+import fs2.Stream.empty
+import fs2.io._
+import fs2.io.file.readRange
+import org.http4s.headers._
+import org.http4s.Status.NotModified
+import org.log4s.getLogger
+
+import scala.concurrent.ExecutionContext
+
+
+trait Static[F[_]] {
+
+  def calcETag(f: File): F[String]
+
+  def fromString(
+    url: String,
+    blockingExecutionContext: ExecutionContext,
+    req: Option[Request[F]] = None): OptionT[F, Response[F]]
+
+  def fromResource(
+    name: String,
+    blockingExecutionContext: ExecutionContext,
+    req: Option[Request[F]] = None,
+    preferGzipped: Boolean = false): OptionT[F, Response[F]]
+
+
+  def fromURL(
```

Comment:
`UrlConnection` gives us enough to calculate a reasonable `Etag`.  Should there be a `UrlConnection => F[Etag]` method, and a parameter?

Reply:
Heck, it'd even work for `fromResource`.

Reply:
@rossabaker I'm unsure of what you mean `fromResource` here.

As for your first comment, I have a very basic understanding of `Etag's` Is there any more info you could give on going from a `UrlConnection => Etag` ?

Reply:
`URLConnection` gives us a lastmodified time and a content length, both of which are often used in hashes of ETag.  Hashing the content makes the best `Etag`, but then you have to load the whole thing, which fights against streaming.  But anything we use to Etag a file should be, for the most part, available here.